aws-distro-opentelemetry-node-autoinstrumentation/test/patches/instrumentation-patch.test.ts

```diff
@@ -692,6 +693,60 @@ describe('InstrumentationPatchTest', () => {
       });
     });
 
+    it('Prevents recursion when credentials provider makes STS calls', async () => {
+      // Track credentials provider calls and skip flag state
+      let credentialsProviderCallCount = 0;
+      let skipCredentialCaptureValue = false;
+
+      // Mock span for attribute capture
+      const mockSpan = { setAttribute: sinon.stub() };
+      sinon.stub(trace, 'getSpan').returns(mockSpan as unknown as Span);
+
+      // Patch ROOT_CONTEXT to capture setValue calls
+      const originalSetValue = ROOT_CONTEXT.setValue;
+      ROOT_CONTEXT.setValue = (key: symbol, value: any) => {
+        if (key.toString().includes('skip-credential-capture')) {
+          skipCredentialCaptureValue = value;
+        }
+        return ROOT_CONTEXT;
+      };
```

Comment:
This changes the actual functionality of `setValue` that we want to test - https://github.com/open-telemetry/opentelemetry-js/blob/48538b46add710483af3ff8147572d5bd4b273e0/api/src/context/context.ts#L50-L54.
However, the real functionality of `setValue` will return a new child Context, so your patch won't be applied to the child OTel Context. Let's just use the original `ROOT_CONTEXT` without patching it.

My understanding is that the `expect(credentialsProviderCallCount).toBe(1)` validation already implies that `expect(skipCredentialCaptureValue).toBe(true)` is validated. So maybe you don't need the `skipCredentialCaptureValue` validation?